### PR TITLE
Feat/new kelly formula

### PR DESCRIPTION
**Pull request by cyberosa**
Improving the kelly formula by removing the impact of confidence

### CODE REVIEW

**Review by Adamantios — CHANGES_REQUESTED** [review]
Looks good, but please give the strategy a more descriptive name, instead of new kelly.

File: packages/valory/services/trader/service.yaml [hunk 1]
```diff
@@ -247,7 +247,7 @@ type: skill
         tool_punishment_multiplier: ${TOOL_PUNISHMENT_MULTIPLIER:int:1}
         redeem_round_timeout: ${REDEEM_ROUND_TIMEOUT:float:3600.0}
         contract_timeout: ${CONTRACT_TIMEOUT:float:300.0}
-        file_hash_to_strategies_json: ${FILE_HASH_TO_STRATEGIES_JSON:list:[["bafybeihufqu2ra7vud4h6g2nwahx7mvdido7ff6prwnib2tdlc4np7dw24",["bet_amount_per_threshold"]],["bafybeif55cu7cf6znyma7kxus4wxa2doarhau2xmndo57iegshxorivwmq",["kelly_criterion"]]]}
+        file_hash_to_strategies_json: ${FILE_HASH_TO_STRATEGIES_JSON:list:[["bafybeihufqu2ra7vud4h6g2nwahx7mvdido7ff6prwnib2tdlc4np7dw24",["bet_amount_per_threshold"]],["bafybeif55cu7cf6znyma7kxus4wxa2doarhau2xmndo57iegshxorivwmq",["kelly_criterion"]],["bafybeihcm4mmf2qqhe3ly47rxtmhcjxrlnnujwyzkcsq6jkbnu6ojxrwja",["new_kelly_criterion"]]]}
```
Comment (Adamantios) [inline packages/valory/services/trader/service.yaml:250]: Remove the old kelly here. Moreover, you are still using the `kelly_criterion` as a trading strategy above.

Comment (Adamantios) [reply packages/valory/services/trader/service.yaml:250]: The hash is incorrect, it points to the executable instead of the custom package.

Comment (cyberosa) [reply packages/valory/services/trader/service.yaml:250]: where is the hash for the custom package?

Comment (Adamantios) [reply packages/valory/services/trader/service.yaml:250]: It is in the output of the command that publishes it to IPFS:

`autonomy ipfs add`

Comment (cyberosa) [reply packages/valory/services/trader/service.yaml:250]: ok found it. Creating a new commit with changes

File: packages/packages.json [hunk 2]
```diff
@@ -4,6 +4,8 @@
         "custom/valory/mike_strat/0.1.0": "bafybeihjiol7f4ch4piwfikurdtfwzsh6qydkbsztpbwbwb2yrqdqf726m",
         "custom/jhehemann/kelly_criterion/0.1.0": "bafybeif55cu7cf6znyma7kxus4wxa2doarhau2xmndo57iegshxorivwmq",
         "custom/w1kke/always_blue/0.1.0": "bafybeieshu32h3es2fslduuhr7nimuvh2vuibyeqdunzrcggaeohekg3jm",
+        "custom/valory/new_kelly_criterion/0.1.0": "bafybeiag5m7nc6fy4vkbz5bmwv25cpkmq7psnfxvdcfktk6eqaun6k5ble",
```
Comment (Adamantios) [inline packages/packages.json:7]:
```suggestion
```

File: packages/valory/customs/kelly_criterion_no_conf/__init__.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2023-2024 Valory AG
```
Comment (Adamantios) [inline packages/valory/customs/kelly_criterion_no_conf/__init__.py:4]:
```suggestion
#   Copyright 2024 Valory AG
```

File: packages/valory/customs/kelly_criterion_no_conf/__init__.py [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""This module contains the new kelly criterion strategy."""
```
Comment (Adamantios) [inline packages/valory/customs/kelly_criterion_no_conf/__init__.py:20]:
```suggestion
"""This module contains the kelly criterion strategy without confidence factor in the
  formula."""
```

File: packages/valory/customs/kelly_criterion_no_conf/kelly_criterion_no_conf.py [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+    )
+    return {"bet_amount": adj_kelly_bet_amount, "info": info, "error": error}
+
+
+def run(*_args, **kwargs) -> Dict[str, Union[int, List[str]]]:
+    """Run the strategy."""
+    missing = check_missing_fields(kwargs)
+    if len(missing) > 0:
+        return {"error": [f"Required kwargs {missing} were not provided."]}
+    print("Executing the NEW KELLY CRITERION FORMULA")
```
Comment (Adamantios) [inline packages/valory/customs/kelly_criterion_no_conf/kelly_criterion_no_conf.py:154]:
```suggestion
```

File: packages/valory/services/trader/service.yaml
Comment (Adamantios) [inline packages/valory/services/trader/service.yaml:1]: https://github.com/valory-xyz/trader/pull/267#discussion_r1628005018

**Review by Adamantios — CHANGES_REQUESTED** [review]

File: packages/packages.json (hunk 2 above)
Comment (cyberosa) [reply packages/packages.json]: fixed

**Review by Adamantios — COMMENTED** [review]
Code looks good, you need to fix the hashes.

**Pull request merged**